src/Blocks/BlockStems.h

```diff
@@ -60,18 +85,28 @@ class cBlockStemsHandler:
 		auto oldMeta = a_Chunk.GetMeta(a_RelPos);
 		auto meta = oldMeta + a_NumStages;
 		a_Chunk.SetBlock(a_RelPos, m_BlockType, static_cast<NIBBLETYPE>(std::min(meta, 7)));  // Update the stem
-		if (meta > 7)
+		if (oldMeta == 7)
```

Comment:
This change is still wrong. The Grow function was fine as it was.

Reply:
I don't really understand what can I do if I don't have to edit this method, here I've tested this method and it works

Reply:
I'll rework this myself and then merge - but can you explain what your reason was for changing this function in the first place?

Reply:
With the old implementation, when you right-click on a melon / pumpkin seed with level = 6, the level of the plant become 7 (full grown) and a melon / pumpkin block appears.
This is because this method checks when level will become > 7.
With this implementation, if old level is before 7 (so 6, 5, ...), no melon / pumpkin block will spawn.

Reply:
Another way is to create a special method when a player / dispenser want to grow melon / pumpkin or add a new parameter to this function, but after some tests it seams this solution works